source/includes/read/count.scala

```diff
@@ -0,0 +1,55 @@
+import org.mongodb.scala._
+import org.mongodb.scala.model.Filters._
+import org.mongodb.scala.model.CountOptions
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object Count {
+  def main(args: Array[String]): Unit = {
+    val uri: String = "<connection string>"
+    val client: MongoClient = MongoClient(uri)
+
+    // start-db-coll
+    val database: MongoDatabase = mongoClient.getDatabase("sample_training")
+    val collection: MongoCollection[Document] = database.getCollection("companies")
+    // end-db-coll
+
+    // Counts all documents in the collection
+    // start-count-all
+    collection.countDocuments().subscribe((count: Long) => println(s"Number of documents: $count"),
+                             (e: Throwable) => println(s"There was an error: $e"))
```

Comment:
S: improve horizontal scroll by newlining where possible or reducing indentation
Applies to all included code snippets